### PR TITLE
BCMOHAD-15683||Defect Fixes

### PR DESCRIPTION
**Pull request by ahshaik1**

### CODE REVIEW

**Review by NataliaNikishina — APPROVED** [review]
@ahshaik1 The branch was successfully validated so I am approving the PR. 
cc: @cgijeffolsen @crystalz2022 @rahulmantriSFDC Please review the PR. Many thanks!

File: TPL/force-app/main/default/classes/HealthcareCostTriggerHandler.cls [hunk 1]
```diff
@@ -7,47 +7,159 @@ public with sharing class HealthcareCostTriggerHandler {
             caseIds.add(hcc.Case2__c);
         }
         System.debug(caseIds);
-        List<AggregateResult> groupedResults = [select SUM(Total_Cost_Override__c)tot,case2__c,RecordType.DeveloperName,sum(Cost__c)costSum from Healthcare_Cost__c where Case2__c IN:caseIds and ((Cost_Include__c = true and Cost_Review__c = true) or (RecordType.DeveloperName IN ('Continuing_Care','Future_Care')))    GROUP BY case2__c,RecordType.DeveloperName] ;
-        System.debug(groupedResults);
+        List<Healthcare_Cost__c> hccSumList = [select id,Total_Cost_Override__c,case2__c,RecordTypeName__c,Cost__c,Cost_Include__c,Cost_Review__c from Healthcare_Cost__c where 
+                                               case2__c IN:caseIds];
+        System.debug(hccSumList);
```
Comment (cgijeffolsen) [inline TPL/force-app/main/default/classes/HealthcareCostTriggerHandler.cls:12]: These retained debug statements will slow down code operation.

File: TPL/force-app/main/default/classes/HealthcareCostTriggerHandler.cls [hunk 2]
```diff
@@ -7,47 +7,159 @@ public with sharing class HealthcareCostTriggerHandler {
...
-               
-            }
-            if((String)agg.get('DeveloperName') == 'Continuing_Care'){
-               
-                c.Continuing_Care_Total2__c = (Decimal)agg.get('costSum');
+            }else{
+                if(caseIdRecordTypeCostMap.containsKey(hcc.Case2__c)){
+                    Map<String,Decimal> tempMap = caseIdRecordTypeCostMap.get(hcc.Case2__c);
+                    if(tempMap.containskey(hcc.RecordTypeName__c)){
+                        System.debug(hcc.RecordTypeName__c);
```
Comment (cgijeffolsen) [inline TPL/force-app/main/default/classes/HealthcareCostTriggerHandler.cls:55]: If you wish to retain the debug cover them with a boolean constant to deactivate them in the class opening statements.

**Review by cgijeffolsen — APPROVED** [review]
1. We will add a ticket to remove all debug logs further in the process as I believe more than this class is affected. @NataliaNikishina 
3. I want to review specific limitation on delete where item has to be Cost Included to affect Totals on delete with the developer (while an edge case, I want to confirm this is covered in the calling class). @ahshaik1

**Review by rahul-mantri — APPROVED** [review]
Validation is successful. So Approving the PR.

**Review by crystalz2022 — APPROVED** [review]
Approved @ahshaik1

**Pull request merged**